crates/node-file-trace/src/lib.rs

```diff
@@ -531,14 +532,18 @@ async fn main_operation(
             let input = process_input(&dir, &context, input).unwrap();
             let fs = create_fs("context directory", &context, watch).await?;
             let mut output_nft_assets = Vec::new();
+            let mut emits = Vec::new();
             for module in input_to_modules(fs, input, process_cwd, exact)
                 .await?
                 .iter()
             {
                 let nft_asset = NftJsonAssetVc::new(*module);
                 let path = nft_asset.path().await?.path.clone();
                 output_nft_assets.push(path);
-                emit(nft_asset.into())
+                emits.push(emit_asset(nft_asset.into()));
+            }
+            for emit in emits {
+                emit.await?;
```

Comment:
Can you add a comment explaining why this is necessary above both the diffs in this file?